frontend/dockerfile/dockerfile_test.go

```diff
@@ -9327,6 +9328,140 @@ COPY Dockerfile /foo
 	}
 }
 
+func testPlatformWithOSVersion(t *testing.T, sb integration.Sandbox) {
```

Comment:
thinking of a workaround to test this on Windows, give we don't have support for `scratch` and the unionfs-like mounting... shouldn't be blocking if it's the only one pending, we can skip on Windows for now and file tracking issue on me.

Reply:
https://github.com/moby/buildkit/issues/5633

Reply:
I suspect, once CI finishes, that this test will fail even on Linux because we don't get `OSVersion` comparisons when `platform.OS` is not `windows`.

Reply:
Maybe a little janky, but perhaps we can have a custom OS platform like "buildkit.test" that will have it use a custom matcher?

Reply:
Better yet, just setting the OS to "windows" instead of "foo" seems to be good enough.